### PR TITLE
Misc improvements and fixes for local setup

### PR DESCRIPTION
**Pull request by chiragjn**
- Fixed exception handling in prisma store
- Add prisma cli during build time
- Refactored providers in sample config
- Fix hosts and ports in compose.env
- Added a workflow for pre-commit

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -29,12 +29,9 @@ RUN if [ "${ADD_PARSER}" = "1" ]; then python3 -m pip install --use-pep517 --no-
 ARG ADD_VECTORDB=0
 RUN if [ "${ADD_VECTORDB}" = "1" ]; then python3 -m pip install --use-pep517 --no-cache-dir -r /tmp/vectordb.requirements.txt; fi
 
-# Temporary addition until templates have been updated using build args as environment variables
-ARG ADD_RERANKER_SVC_URL=""
```
Comment (S1LV3RJ1NX) [inline backend/Dockerfile:33]: Can we keep this, reranker is still required when deploying via templates, as there is no way to inject env vars at the moment.

File: compose.env [hunk 2]
```diff
@@ -17,14 +17,14 @@ INFINITY_RERANKING_MODEL=mixedbread-ai/mxbai-rerank-xsmall-v1
...
-INFINITY_URL=http://infinity-server:7797
+INFINITY_URL=http://infinity-server:7997
 COGNITA_BACKEND_PORT=8000
 METADATA_STORE_CONFIG='{"provider":"prisma"}'
 VECTOR_DB_CONFIG='{"provider":"qdrant","url":"http://qdrant-server:6333", "config": {"grpc_port": 6334, "prefer_grpc": false}}'
 
 ## COGNITA_FRONTEND VARS
 COGNITA_FRONTEND_PORT=5001
-VITE_QA_FOUNDRY_URL=http://localhost:8000
+VITE_QA_FOUNDRY_URL=http://cognita-backend:8000
```
Comment (S1LV3RJ1NX) [inline compose.env:27]: Front-end for some reason is not able to connect to `cognit-backend` hence had to add localhost.

File: docker-compose.yaml [hunk 3]
```diff
@@ -128,9 +139,9 @@ services:
             - TFY_API_KEY=${TFY_API_KEY}
             - TFY_HOST=${TFY_HOST}
             - TFY_LLM_GATEWAY_URL=${TFY_LLM_GATEWAY_URL}
-            - DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@db:${POSTGRES_PORT}/cognita-config
+            - DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@cognita-db:5432/cognita-config
```
Comment (S1LV3RJ1NX) [inline docker-compose.yaml:142]: linux systems are not able to access db via service name. #244

File: backend/Dockerfile (hunk 1 above)
Comment (chiragjn) [reply backend/Dockerfile:33]: Simply setting it in env var in TrueFoundry spec should work?

File: compose.env (hunk 2 above)
Comment (chiragjn) [reply compose.env:27]: Why? It should be able to connect, they are part of the same bridge network

I don't understand how will it work via localhost, unless frontend has requirement to talk to backend via host network.
Need to understand what breaks here

File: docker-compose.yaml (hunk 3 above)
Comment (chiragjn) [reply docker-compose.yaml:142]: Why? docker dns is supposed to work for bridge network on any OS
We should reproduce and confirm

File: compose.env (hunk 2 above)
Comment (chiragjn) [reply compose.env:27]: I am able to use the UI just fine with these settings

Comment (S1LV3RJ1NX) [reply compose.env:27]: Intresting, we can check on this... @sayan-truefoundry and I had issues.

File: backend/Dockerfile (hunk 1 above)
Comment (chiragjn) [reply backend/Dockerfile:33]: I'll revert this but this is not the correct solution
Any change requires a rebuild

**Review by S1LV3RJ1NX — APPROVED** [review]

**Pull request merged**